### PR TITLE
Cmm azurebadge

### PR DESCRIPTION
**Pull request by chmorgan**

### CODE REVIEW

**Comment by chmorgan** [comment]
@kayoub5 well the good news is that most of the builds look to be working reasonably well, the blocker is the send tests not receiving the packets on the receive in almost all cases

**Comment by kayoub5** [comment]
@chmorgan Remote tests do not work with npcap

File: SharpPcap/LibPcap/SendQueue.cs [hunk 1]
```diff
@@ -152,10 +152,8 @@ protected unsafe int ManagedTransmit(PcapDevice device, bool synchronized)
                     if (synchronized)
                     {
                         var timestamp = TimeSpan.FromTicks(header.Date.Ticks);
-                        while (sw.Elapsed < timestamp - firstTimestamp)
-                        {
-                            // Wait for packet time
-                        }
+                        var remainingTime = timestamp.Subtract(firstTimestamp);
+                        System.Threading.Thread.Sleep(remainingTime);
```
Comment (kayoub5) [inline SharpPcap/LibPcap/SendQueue.cs]: @chmorgan This usually gives more accurate results on windows, could you give it a try on mac?
```suggestion
                        while (sw.Elapsed < timestamp - firstTimestamp)
                        {
                              // Wait for packet time
                              Thread.Sleep((timestamp - firstTimestamp)/ 2);
                        }
```

Comment (chmorgan) [reply SharpPcap/LibPcap/SendQueue.cs]: @kayoub5 I couldn't find any numbers on sleep accuracy either way. How do you know that this timing gives better results?

Comment (kayoub5) [reply SharpPcap/LibPcap/SendQueue.cs]: This is the best method we found that does not consume too much CPU (running some timer logic with millisecond precision).

For Micro/Nano second precision, the best thing I found was (this method consumes much more CPU than sleep)
https://stackoverflow.com/a/6259726/1438522

File: SharpPcap/LibPcap/PcapDeviceCaptureLoop.cs [hunk 2]
```diff
@@ -197,10 +197,10 @@ public virtual void StopCapture()
         {
             if (Started)
             {
-                shouldCaptureThreadStop = true;
+                threadCancellationTokenSource.Cancel();
+                LibPcapSafeNativeMethods.pcap_breakloop(PcapHandle);
```
[the remark this answers is not in the document]
Comment (chmorgan) [reply SharpPcap/LibPcap/PcapDeviceCaptureLoop.cs]: Yep, it sounds like that's what the current behavior could be. Can't wait for a new libpcap with that pcap_breakloop support!

File: SharpPcap/LibPcap/SendQueue.cs (hunk 1 above)
Comment (chmorgan) [reply SharpPcap/LibPcap/SendQueue.cs]: Updated the changes to sleep with the incremental delay as suggested.

**Pull request merged**